app/src/main/java/fr/free/nrw/commons/upload/SingleUploadFragment.java

```diff
@@ -104,13 +105,75 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         Log.d("Single Upload fragment", license);
 
-        ArrayAdapter<String> adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_dropdown_item, licenseItems);
+        ArrayAdapter<String> adapter;
+        if (PreferenceManager.getDefaultSharedPreferences(getActivity()).getBoolean("theme",true)) {
+            // dark theme
+            adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_dropdown_item, licenseItems);
+        }else {
+            // light theme
+            adapter = new ArrayAdapter<>(getActivity(), R.layout.light_simple_spinner_dropdown_item, licenseItems);
+        }
+
         licenseSpinner.setAdapter(adapter);
 
         int position = licenseItems.indexOf(getString(Utils.licenseNameFor(license)));
         Log.d("Single Upload fragment", "Position:"+position+" "+getString(Utils.licenseNameFor(license)));
         licenseSpinner.setSelection(position);
 
+        licenseSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+
+                //Set selected color to white because it should be readable on random images.
```

Comment:
Could you put these changes in the event listener for item selected thats already defined below. :)